packages/fuels/src/index.test.ts

```diff
@@ -1,5 +1,40 @@
+/* eslint-disable no-cond-assign */
+import * as fs from 'fs';
+import * as prettier from 'prettier';
+
 import * as fuels from './index';
 
+// this function parses a bundled file and returns a list of all modules that it references (import, require, export from, etc.)
+const getListOfModules = (bundleContent: string) => {
+  const formattedBundleContent = prettier.format(bundleContent, {
+    parser: 'babel',
+  });
+  const listOfModules = new Set<string>();
+
+  const requireRegex = /(?:^|[^.])\brequire\s*\(\s*(["'])(.+?)\1\s*\)/g;
+  let match;
+  while ((match = requireRegex.exec(formattedBundleContent))) {
+    const moduleName = match[2];
+    listOfModules.add(moduleName);
+  }
+
+  const importRegex = /import\s+(?:.+\s+from\s+)?["'](.+)["'];?/g;
+  let importMatch;
+  while ((importMatch = importRegex.exec(formattedBundleContent))) {
+    const moduleName = importMatch[1];
+    listOfModules.add(moduleName);
+  }
+
+  const exportRegex = /export\s+(?:.+\s+from\s+)?["'](.+)["'];?/g;
+  let exportMatch;
+  while ((exportMatch = exportRegex.exec(formattedBundleContent))) {
+    const moduleName = exportMatch[1];
+    listOfModules.add(moduleName);
+  }
```

Comment:
The approach that I have gone with is scanning the bundled files for `require` and `import` statements to check for any references to Node.js built-in modules that are only available in a Node environment and not in a browser. I am not sure if this is the absolute best way to go about this, but I have checked that it _does_ get the job done.

Reply:
The rationale is valid, but it'd be hard to have certainty about the complete required stack, since we don't control all the libraries in use. For example, we can import an NPM package that uses some native module, which might be hard or impossible to detect, especially if we consider multiple nested dependencies.

In the end, I believe a declarative approach will suit us better.

One inspiration is how [jest-runner-groups](https://github.com/eugene-manuilov/jest-runner-groups) does it using manual tags.